// @storybook/addon-a11y/dist/components/Report/index.js
"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("core-js/modules/es6.array.map");

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _components = require("@storybook/components");

var _Item = _interopRequireDefault(require("./Item"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var Report = function Report(_ref) {
  var items = _ref.items,
      empty = _ref.empty,
      passes = _ref.passes;
  return _react.default.createElement(_react.Fragment, null, items.length ? items.map(function (item) {
    return _react.default.createElement(_Item.default, {
      passes: passes,
      item: item,
      key: item.id
    });
  }) : _react.default.createElement(_components.Placeholder, {
    key: "placeholder"
  }, empty));
};

Report.propTypes = {
  items: _propTypes.default.arrayOf(_propTypes.default.shape({
    description: _propTypes.default.string,
    nodes: _propTypes.default.array,
    tags: _propTypes.default.array
  })).isRequired,
  empty: _propTypes.default.string.isRequired,
  passes: _propTypes.default.bool.isRequired
};
var _default = Report;
exports.default = _default;
