// @storybook/addon-a11y/dist/components/Report/Info.js
"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _theming = require("@storybook/theming");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Wrapper = _theming.styled.div(function (_ref) {
  var theme = _ref.theme;
  return {
    backgroundColor: theme.background.bar,
    padding: '12px',
    marginBottom: '10px'
  };
});

var Help = _theming.styled.p({
  margin: '0 0 12px'
});

var Link = _theming.styled.a({
  marginTop: '12px',
  textDecoration: 'underline',
  color: 'inherit',
  display: 'block'
});

function Info(_ref2) {
  var item = _ref2.item;
  return _react.default.createElement(Wrapper, null, _react.default.createElement(Help, null, item.help), _react.default.createElement(Link, {
    href: item.helpUrl,
    target: "_blank"
  }, "More info..."));
}

Info.propTypes = {
  item: _propTypes.default.shape({
    help: _propTypes.default.node,
    helpUrl: _propTypes.default.string
  }).isRequired
};
var _default = Info;
exports.default = _default;
