// @storybook/addon-a11y/dist/components/Report/Rules.js
"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("core-js/modules/es6.array.map");

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _theming = require("@storybook/theming");

var _components = require("@storybook/components");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var impactColors = {
  minor: '#f1c40f',
  moderate: '#e67e22',
  serious: '#e74c3c',
  critical: '#c0392b',
  success: '#2ecc71'
};

var List = _theming.styled.div({
  display: 'flex',
  flexDirection: 'column',
  padding: '4px',
  fontWeight: '400'
});

var Item = _theming.styled.div({
  display: 'flex',
  flexDirection: 'row',
  marginBottom: '6px'
});

var Message = _theming.styled.div({
  paddingLeft: '6px'
});

var Status = _theming.styled.div(function (_ref) {
  var passes = _ref.passes,
      impact = _ref.impact;
  return {
    height: '16px',
    width: '16px',
    borderRadius: '8px',
    fontSize: '10px',
    display: 'inline-flex',
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
    flex: '0 0 16px',
    color: passes ? impactColors.success : impactColors[impact]
  };
});

var Rule = function Rule(_ref2) {
  var rule = _ref2.rule,
      passes = _ref2.passes;
  return _react.default.createElement(Item, null, _react.default.createElement(Status, {
    passes: passes || undefined,
    impact: rule.impact
  }, passes ? _react.default.createElement(_components.Icons, {
    icon: "check"
  }) : _react.default.createElement(_components.Icons, {
    icon: "cross"
  })), _react.default.createElement(Message, null, rule.message));
};

Rule.propTypes = {
  rule: _propTypes.default.shape({
    message: _propTypes.default.node
  }).isRequired,
  passes: _propTypes.default.bool.isRequired
};
/* eslint-disable react/no-array-index-key */

function Rules(_ref3) {
  var rules = _ref3.rules,
      passes = _ref3.passes;
  return _react.default.createElement(List, null, rules.map(function (rule, index) {
    return _react.default.createElement(Rule, {
      passes: passes,
      rule: rule,
      key: index
    });
  }));
}

Rules.propTypes = {
  rules: _propTypes.default.arrayOf(_propTypes.default.shape({
    message: _propTypes.default.node
  })).isRequired,
  passes: _propTypes.default.bool.isRequired
};
var _default = Rules;
exports.default = _default;
