// @storybook/addon-info/dist/components/markdown/code.js
"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Blockquote = Blockquote;
Object.defineProperty(exports, "Pre", {
  enumerable: true,
  get: function get() {
    return _pre.default;
  }
});
exports.Code = void 0;

require("core-js/modules/es6.object.assign");

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _components = require("@storybook/components");

var _pre = _interopRequireDefault(require("./pre/pre"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

// XXX: is this a bug? should it be (props) => ?
var Code = function Code(_ref) {
  var props = _ref.props;
  return _react.default.createElement(_components.SyntaxHighlighter, _extends({
    bordered: true,
    copyable: true
  }, props));
};

exports.Code = Code;
Code.propTypes = {
  props: _propTypes.default.shape({}).isRequired
};

function Blockquote(_ref2) {
  var children = _ref2.children;
  var style = {
    fontSize: '1.88em',
    fontFamily: 'Menlo, Monaco, "Courier New", monospace',
    borderLeft: '8px solid #fafafa',
    padding: '1rem'
  };
  return _react.default.createElement("blockquote", {
    style: style
  }, children);
}

Blockquote.propTypes = {
  children: _propTypes.default.node
};
Blockquote.defaultProps = {
  children: null
};
