// @storybook/addon-a11y/dist/components/Panel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("core-js/modules/es7.symbol.async-iterator");

require("core-js/modules/es6.symbol");

require("core-js/modules/es6.object.define-property");

require("core-js/modules/es6.object.create");

require("core-js/modules/es6.object.set-prototype-of");

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _theming = require("@storybook/theming");

var _coreEvents = require("@storybook/core-events");

var _components = require("@storybook/components");

var _constants = _interopRequireDefault(require("../constants"));

var _Tabs = _interopRequireDefault(require("./Tabs"));

var _Report = _interopRequireDefault(require("./Report"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var Icon = (0, _theming.styled)(_components.Icons)({
  height: '12px',
  width: '12px',
  marginRight: '4px'
}, function (_ref) {
  var status = _ref.status,
      theme = _ref.theme;
  return status === 'running' ? {
    animation: "".concat(theme.animation.rotate360, " 1s linear infinite;")
  } : {};
});

var Passes = _theming.styled.span(function (_ref2) {
  var theme = _ref2.theme;
  return {
    color: theme.color.positive
  };
});

var Violations = _theming.styled.span(function (_ref3) {
  var theme = _ref3.theme;
  return {
    color: theme.color.negative
  };
});

var A11YPanel =
/*#__PURE__*/
function (_Component) {
  _inherits(A11YPanel, _Component);

  function A11YPanel() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, A11YPanel);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(A11YPanel)).call.apply(_getPrototypeOf2, [this].concat(args)));
    _this.state = {
      status: 'ready',
      passes: [],
      violations: []
    };

    _this.onUpdate = function (_ref4) {
      var passes = _ref4.passes,
          violations = _ref4.violations;

      _this.setState({
        status: 'ran',
        passes: passes,
        violations: violations
      }, function () {
        setTimeout(function () {
          var status = _this.state.status;

          if (status === 'ran') {
            _this.setState({
              status: 'ready'
            });
          }
        }, 900);
      });
    };

    _this.request = function () {
      var _this$props = _this.props,
          api = _this$props.api,
          active = _this$props.active;

      if (active) {
        _this.setState({
          status: 'running'
        }, function () {
          api.emit(_constants.default.REQUEST);
        });
      }
    };

    return _this;
  }

  _createClass(A11YPanel, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var api = this.props.api;
      api.on(_coreEvents.STORY_RENDERED, this.request);
      api.on(_constants.default.RESULT, this.onUpdate);
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      // TODO: might be able to remove this
      var active = this.props.active;

      if (!prevProps.active && active) {
        this.request();
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      var api = this.props.api;
      api.off(_coreEvents.STORY_RENDERED, this.request);
      api.off(_constants.default.RESULT, this.onUpdate);
    }
  }, {
    key: "render",
    value: function render() {
      var _this$state = this.state,
          passes = _this$state.passes,
          violations = _this$state.violations,
          status = _this$state.status;
      var active = this.props.active;
      var actionTitle;

      if (status === 'ready') {
        actionTitle = 'Rerun tests';
      } else if (status === 'running') {
        actionTitle = _react.default.createElement(_react.Fragment, null, _react.default.createElement(Icon, {
          inline: true,
          icon: "sync",
          status: status
        }), " Running test");
      } else if (status === 'ran') {
        actionTitle = _react.default.createElement(_react.Fragment, null, _react.default.createElement(Icon, {
          inline: true,
          icon: "check"
        }), " Tests completed");
      }

      return active ? _react.default.createElement(_react.Fragment, null, _react.default.createElement(_Tabs.default, {
        key: "tabs",
        tabs: [{
          label: _react.default.createElement(Violations, null, violations.length, " Violations"),
          panel: _react.default.createElement(_Report.default, {
            passes: false,
            items: violations,
            empty: "No a11y violations found."
          })
        }, {
          label: _react.default.createElement(Passes, null, passes.length, " Passes"),
          panel: _react.default.createElement(_Report.default, {
            passes: true,
            items: passes,
            empty: "No a11y check passed"
          })
        }]
      }), _react.default.createElement(_components.ActionBar, {
        key: "actionbar",
        actionItems: [{
          title: actionTitle,
          onClick: this.request
        }]
      })) : null;
    }
  }]);

  return A11YPanel;
}(_react.Component);

A11YPanel.propTypes = {
  active: _propTypes.default.bool.isRequired,
  api: _propTypes.default.shape({
    on: _propTypes.default.func,
    emit: _propTypes.default.func,
    off: _propTypes.default.func
  }).isRequired
};
var _default = A11YPanel;
exports.default = _default;
