// @storybook/addon-a11y/dist/index.js
"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.configureA11y = exports.checkA11y = exports.withA11Y = exports.withA11y = void 0;

require("core-js/modules/es6.object.define-properties");

require("core-js/modules/es6.object.freeze");

require("core-js/modules/es6.promise");

var _global = require("global");

var _axeCore = _interopRequireDefault(require("axe-core"));

var _utilDeprecate = _interopRequireDefault(require("util-deprecate"));

var _commonTags = require("common-tags");

var _addons = _interopRequireDefault(require("@storybook/addons"));

var _coreEvents = require("@storybook/core-events");

var _constants = _interopRequireWildcard(require("./constants"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    configureA11y is deprecated, please configure addon-a11y using the addParameter api:\n    \n    addParameters({\n      a11y: {\n        // ... axe options\n        element: '#root', // optional selector which element to inspect\n      },\n    });\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var channel = _addons.default.getChannel();

var progress = Promise.resolve();
var setup = {};

var getElement = function getElement() {
  var storyRoot = _global.document.getElementById('story-root');

  if (storyRoot) {
    return storyRoot.children;
  }

  return _global.document.getElementById('root');
};

var report = function report(input) {
  channel.emit(_constants.default.RESULT, input);
};

var run = function run(config, options) {
  progress = progress.then(function () {
    _axeCore.default.reset();

    if (config) {
      _axeCore.default.configure(config);
    }

    return _axeCore.default.run(getElement(), options || {
      restoreScroll: true
    }).then(report);
  });
}; // NOTE: we should add paramaters to the STORY_RENDERED event and deprecate this


var withA11y = function withA11y(getStory, context) {
  var params = context.parameters[_constants.PARAM_KEY];

  if (params) {
    setup = params;
  }

  return getStory(context);
};

exports.withA11y = withA11y;
channel.on(_coreEvents.STORY_RENDERED, function () {
  return run(setup.config, setup.options);
});
channel.on(_constants.default.REQUEST, function () {
  return run(setup.config, setup.options);
});

if (module && module.hot && module.hot.decline) {
  module.hot.decline();
} // TODO: REMOVE at v6.0.0


var withA11Y = (0, _utilDeprecate.default)(function () {
  return withA11y.apply(void 0, arguments);
}, 'withA11Y has been renamed withA11y'); // TODO: REMOVE at v6.0.0

exports.withA11Y = withA11Y;
var checkA11y = (0, _utilDeprecate.default)(function () {
  return withA11y.apply(void 0, arguments);
}, 'checkA11y has been renamed withA11y'); // TODO: REMOVE at v6.0.0

exports.checkA11y = checkA11y;
var configureA11y = (0, _utilDeprecate.default)(function (config) {
  setup = config;
}, (0, _commonTags.stripIndents)(_templateObject()));
exports.configureA11y = configureA11y;
