// @storybook/addon-storyshots/dist/frameworks/angular/helpers.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
// tslint:disable-next-line:no-implicit-dependencies
var core_1 = require("@angular/core");
// tslint:disable-next-line:no-implicit-dependencies
var forms_1 = require("@angular/forms");
// tslint:disable-next-line:no-implicit-dependencies
var platform_browser_1 = require("@angular/platform-browser");
var app_component_1 = require("./app.component");
var app_token_1 = require("./app.token");
var getModuleMeta = function (declarations, entryComponents, bootstrap, data, moduleMetadata) {
    return {
        declarations: declarations.concat((moduleMetadata.declarations || [])),
        imports: [platform_browser_1.BrowserModule, forms_1.FormsModule].concat((moduleMetadata.imports || [])),
        providers: [{ provide: app_token_1.STORY, useValue: __assign({}, data) }].concat((moduleMetadata.providers || [])),
        entryComponents: entryComponents.concat((moduleMetadata.entryComponents || [])),
        schemas: (moduleMetadata.schemas || []).slice(),
        bootstrap: bootstrap.slice(),
    };
};
var createComponentFromTemplate = function (template) {
    var componentClass = /** @class */ (function () {
        function DynamicComponent() {
        }
        return DynamicComponent;
    }());
    return core_1.Component({
        template: template,
    })(componentClass);
};
exports.initModuleData = function (storyObj) {
    var component = storyObj.component, template = storyObj.template, props = storyObj.props, _a = storyObj.moduleMetadata, moduleMetadata = _a === void 0 ? {} : _a;
    var AnnotatedComponent = template ? createComponentFromTemplate(template) : component;
    var story = {
        component: AnnotatedComponent,
        props: props,
    };
    var moduleMeta = getModuleMeta([app_component_1.AppComponent, AnnotatedComponent], [AnnotatedComponent], [app_component_1.AppComponent], story, moduleMetadata);
    return {
        AppComponent: app_component_1.AppComponent,
        moduleMeta: moduleMeta,
    };
};
